src/gitlabcache.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Roger Meier <roger@bufferoverflow.ch>
+// SPDX-License-Identifier: MIT
+
+import Crypto from 'crypto';
+
+import {Logger} from '@verdaccio/types';
+import NodeCache from 'node-cache';
+
+export class GitlabCache {
+  private logger: Logger;
+  private ttl: number;
+  private storage: NodeCache;
+
+  public static get DEFAULT_TTL() {
+    return 300;
+  }
+
+  private static _generateKeyHash(username: string, password: string) {
+    const sha = Crypto.createHash('sha256');
+    sha.update(JSON.stringify({ username: username, password: password }));
```

Comment:
**Tip:** You can directly use `sha.update(JSON.stringify({ username, password }));` since `username` and `password` properties shares the same name...